src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Data.cs

```diff
@@ -22,14 +23,28 @@ public sealed partial class PlatformCompatibilityAnalyzer
         ///  - UnsupportedFirst - keeps the lowest version of [UnsupportedOSPlatform] attribute found
         ///  - UnsupportedSecond - keeps the second lowest version of [UnsupportedOSPlatform] attribute found
         /// </summary>
-        private class PlatformAttributes
+        private class Versions
         {
             public Version? SupportedFirst { get; set; }
             public Version? SupportedSecond { get; set; }
             public Version? UnsupportedFirst { get; set; }
             public Version? UnsupportedSecond { get; set; }
-            public bool HasAttribute() => SupportedFirst != null || UnsupportedFirst != null ||
+            public bool IsSet() => SupportedFirst != null || UnsupportedFirst != null ||
                         SupportedSecond != null || UnsupportedSecond != null;
         }
+
+        private sealed class PlatformAttributes
+        {
+            public PlatformAttributes() { }
+
+            public PlatformAttributes(Callsite callsite, SmallDictionary<string, Versions> platforms)
+            {
+                Callsite = callsite;
+                Platforms = platforms;
+            }
+
+            public SmallDictionary<string, Versions>? Platforms { get; set; }
```

Comment:
Is this property ever set to null via the property setter? Both the assignment sites I see below seem to guarantee the value assigned will be non-null. If that is true, I'd prefer this field be made non-nullable so all the read sites for the property don't have to add a `!`

Reply:
By default, the empty constructor used for creating the attributes, and if there were no attributes found for the symbol this property will never be set. https://github.com/dotnet/roslyn-analyzers/blob/8aee460a17c3f05eca78305a00725e9480316fa9/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs#L1392